prototyping/dscp-lang-flows/hyproof-token-flows.dscp

```diff
@@ -0,0 +1,42 @@
+/**
+ * ------------------
+ * Token Declarations
+ * ------------------
+ *
+ * Note that `token`s define formats that should be persisted in transitions.
+ * Fields can be of type Role, File, Literal, a specific literal, a token type
+ * or a union of any of these
+ */
+
+token InitiatedCert {
+  hydrogen_owner: Role,
```

Comment:
I think we need to have an `energy_owner` Role here as well

Reply:
Is this ok ?

Reply:
why the change of name?

Reply:
It makes more sense .. it is like i am the H producer and i would like to say that this person will most likely be the person who's gonna add the embodied_co2 to info to my Cert but that might not happen at all especially if i draft token gets burned / canceled / revoked ( by whatever authority / user )

Reply:
but it's not mutable. So I don't see it as more accurate. I'm asserting this is who produced the energy

Reply:
Up to you , I am just saying , in my mind **`proposed_energy_owner: Role,`** is equal to the owner of the token that will be probably spawned in the future from this one might be EnergyX **OR** null ( null = there is no future because it is not gonna go to step 2 because of cancelations ). While **`energy_owner: Role`** is more suitable when I am sure that's gonna happen no matter what. It is like in a DAO i will use the terminology _proposed_spender / possible_spender_ because I am not sure if that's gonna happen because at any point the DAO community might want to dissolve the DAO. But it is totally up to you, I am not that strict about this key-value pair ...

Reply:
this is just terminology so I don't really care. This just makes property mapping slightly more difficult in the API. You seem to have a very strong opinion and I still don't understand why

Reply:
In the real world there is a contractual relationship between the hydrogen producer and their energy provider. There is no 'proposed' part to it - it's a pre-existing contractual relationship. **Only** _that specific_ energy provider will be allowed to countersign and complete the certificate.